### PR TITLE
Spiked phix tag2

### PR DESCRIPTION
**Pull request by srl147**

### CODE REVIEW

File: lib/npg_pipeline/cache/barcodes.pm [hunk 1]
```diff
@@ -15,12 +15,10 @@ with 'WTSI::DNAP::Utilities::Loggable';
...
-# we used to set both the i7 and i5 spiked phix tags for dual index runs
-Readonly::Scalar my $SPIKED_PHIX_PADDED_2        => q{ACAACGCAATC-TCTTTCCC};
-Readonly::Scalar my $SPIKED_PHIX_I5OPPOSITE_PADDED_2 => q{ACAACGCAATC-AGATCTCG};
-# now we simply add the missing i5 spiked phix tag for dual index runs
-Readonly::Scalar my $SPIKED_PHIX_TAG2        => q{TCTTTCCC};
-Readonly::Scalar my $SPIKED_PHIX_I5OPPOSITE_TAG2 => q{AGATCTCG};
+# For dual index runs add the expected i5 tag sequences for samples (e.g. phix) which have no i5 tag
+# I've extended these to 10 bases but I have not confirmed this is correct for I5OPPOSITE
+Readonly::Scalar my $SPIKED_PHIX_TAG2        => q{TCTTTCCCTA};
+Readonly::Scalar my $SPIKED_PHIX_I5OPPOSITE_TAG2 => q{AGATCTCGTA};
```
Comment (dkj) [inline lib/npg_pipeline/cache/barcodes.pm:21]:
```suggestion
Readonly::Scalar my $SPIKED_PHIX_I5OPPOSITE_TAG2 => q{AGATCTCGGT};
```
From looking for rev comp of `AGATCTCG`, `CGAGATCT` in adapters [doc](https://emea.support.illumina.com/content/dam/illumina-support/documents/documentation/chemistry_documentation/experiment-design/illumina-adapter-sequences-1000000002694-13.pdf) ...

File: lib/npg_pipeline/cache/barcodes.pm [hunk 2]
```diff
@@ -15,12 +15,10 @@ with 'WTSI::DNAP::Utilities::Loggable';
...
 
 Readonly::Scalar my $TAG_LIST_FILE_HEADER      => qq{barcode_sequence\tbarcode_name\tlibrary_name\tsample_name\tdescription};
-# we used to set both the i7 and i5 spiked phix tags for dual index runs
-Readonly::Scalar my $SPIKED_PHIX_PADDED_2        => q{ACAACGCAATC-TCTTTCCC};
-Readonly::Scalar my $SPIKED_PHIX_I5OPPOSITE_PADDED_2 => q{ACAACGCAATC-AGATCTCG};
-# now we simply add the missing i5 spiked phix tag for dual index runs
-Readonly::Scalar my $SPIKED_PHIX_TAG2        => q{TCTTTCCC};
-Readonly::Scalar my $SPIKED_PHIX_I5OPPOSITE_TAG2 => q{AGATCTCG};
+# For dual index runs add the expected i5 tag sequences for samples (e.g. phix) which have no i5 tag
+# I've extended these to 10 bases but I have not confirmed this is correct for I5OPPOSITE
```
Comment (dkj) [inline lib/npg_pipeline/cache/barcodes.pm:19]:
```suggestion
# I've extended these to 10 bases. For I5OPPOSITE final `GT` predicted from adapter documentation.
```

Comment (dkj) [inline lib/npg_pipeline/cache/barcodes.pm:19]:
```suggestion
# I've extended these to 10 bases. For I5OPPOSITE final `GT` predicted from adapter documentation.
```

File: lib/npg_pipeline/cache/barcodes.pm (hunk 1 above)
Comment (dkj) [inline lib/npg_pipeline/cache/barcodes.pm:21]:
```suggestion
Readonly::Scalar my $SPIKED_PHIX_I5OPPOSITE_TAG2 => q{AGATCTCGGT};
```
From looking for rev comp of `AGATCTCG`, `CGAGATCT` in adapters [doc](https://emea.support.illumina.com/content/dam/illumina-support/documents/documentation/chemistry_documentation/experiment-design/illumina-adapter-sequences-1000000002694-13.pdf) ...

**Comment by srl147** [comment]
David


Thanks, I'd completely forgotten I'd actually got round to creating a pull request for this.


I've accepted your changes. I did have a look for a run with 2x10-base tags which ran on a HiSeqX or a HiSeq 4000 but I couldn't find any, all the Visium runs have been on a NovaSeq. I guess we'll have to see what happens when the NovaSeqs are updated and start sequencing the i5 tag in the opposite direction. But we'll need to change the pipeline to handle these anyway.


Steven

________________________________

**Pull request merged**